### PR TITLE
feat: Improve docs for `DataGrid`, export as unstable

### PR DESCRIPTION
**Pull request by ling1726**
Updates the documentation for `DataGrid` and exports the components from `@fluentui/react-components/unstable`

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `62.975 kB`<br />`17.687 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `190.411 kB`<br />`53.077 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |   `33.48 kB`<br />`11.037 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/f7d7f82f6b3117ce163c6fee84b03c8774bed95f'>f7d7f82f6b3117ce163c6fee84b03c8774bed95f</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/Avatar-mount_base.html">1574</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/Avatar-mount.html">1556</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/Button-mount_base.html">1122</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/Button-mount.html">1125</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/FluentProvider-mount_base.html">1863</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/FluentProvider-mount.html">1857</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/FluentProviderWithTheme-mount_base.html">755</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/FluentProviderWithTheme-mount.html">744</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">700</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">699</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">750</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">736</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/MakeStyles-mount_base.html">2257</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/MakeStyles-mount.html">2255</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/SpinButton-mount_base.html">3034</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25805/perf-test-react-components/SpinButton-mount.html">3054</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 1 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-tool-fluent/index.html?organization=uifabric&project=fabricpublic&prId=25805&commitId=3aef303b619db8c1a8dbf2753dbd62a3db12f607&env=prod)

✅ There was 0 screenshots added, 0 screenshots removed, 1750 screenshots unchanged, 0 screenshots with different dimensions and 1 screenshots with visible difference.


  <details><summary><b>unknown</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.size+active+ring-shadow.normal.chromium.png|6| Changed|

</details>


<div id="vrtCommentfluentuiv9"/>

**Review by layershifter — APPROVED** [review]

File: packages/react-components/react-table/stories/DataGrid/DataGridDescription.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<!-- Don't allow prettier to collapse code block into single line -->
+<!-- prettier-ignore -->
+> **⚠️ Preview components are considered unstable:**
+>
+> ```jsx
+>
+> import { Table } from '@fluentui/react-components/unstable';
```
Comment (layershifter) [inline packages/react-components/react-table/stories/DataGrid/DataGridDescription.md:7]: Should be `DataGrid`. IDK why my previous comment is not visible 😕

**Pull request merged**